raiden/utils/nursery.py

```diff
@@ -83,12 +83,18 @@ def subprocess_stopped(result: AsyncResult) -> None:
                         janitor._processes.remove(process)
 
                         # if the subprocess error'ed propagate the error.
-                        if result.get() != STATUS_CODE_FOR_SUCCESS:
-                            log.error("Process died! Bailing out.", args=process.args)
-                            janitor._stop.set()
+                        exit_code = result.get()
```

Comment:
if `result` contains an exception this will raise. The exception has to be handled and then set on the nursery's async result.

Reply:
Well spotted, I added handling for exceptions in `result`